ldm/models/diffusion/dynamic_slice_attention_processor.py

```diff
@@ -84,6 +83,13 @@ def __call__(self, attn: CrossAttention, hidden_states, encoder_hidden_states=No
 
             hidden_states[start_idx:end_idx] = attn_slice
 
+            del query_slice
+            del key_slice
+            del attn_mask_slice
+            del attn_slice
+            if hidden_states.device.type == 'cuda':
+                torch.cuda.empty_cache()
```

Comment:
this has a performance penalty - i had something like it in an earlier version and lots of folks complained so i had to take it out. does it need to be run many times per step, or is just once ok?

Reply:
I tried removing that block in my testing. If I don't clear the cache out right there, it dies. I'm happy to test out an alternative if you share a diff. Maybe one idea is to check if we're over a high-water mark and then empty the cache?

Reply:
This is the classic tradeoff between memory and performance...